src/bindings/webgl/rendering_context.cpp

```diff
@@ -1484,6 +1496,110 @@ namespace webgl
     return env.Undefined();
   }
 
+  Napi::Value WebGL2RenderingContext::VertexAttribI4i(const Napi::CallbackInfo &info)
+  {
+    Napi::Env env = info.Env();
+    Napi::HandleScope scope(env);
+
+    if (info.Length() < 5)
+    {
+      Napi::TypeError::New(env, "vertexAttribI4i() takes 5 arguments: index, x, y, z, w.")
+        .ThrowAsJavaScriptException();
+      return env.Undefined();
+    }
+
+    auto index = info[0].As<Napi::Number>().Uint32Value();
+    auto x = info[1].As<Napi::Number>().Int32Value();
+    auto y = info[2].As<Napi::Number>().Int32Value();
+    auto z = info[3].As<Napi::Number>().Int32Value();
+    auto w = info[4].As<Napi::Number>().Int32Value();
+
+    glContext_->vertexAttribI4i(static_cast<int>(index), x, y, z, w);
+    return env.Undefined();
+  }
+
+  Napi::Value WebGL2RenderingContext::VertexAttribI4ui(const Napi::CallbackInfo &info)
+  {
+    Napi::Env env = info.Env();
+    Napi::HandleScope scope(env);
+
+    if (info.Length() < 5)
+    {
+      Napi::TypeError::New(env, "vertexAttribI4ui() takes 5 arguments: index, x, y, z, w.")
+        .ThrowAsJavaScriptException();
+      return env.Undefined();
+    }
+
+    auto index = info[0].As<Napi::Number>().Uint32Value();
+    auto x = info[1].As<Napi::Number>().Uint32Value();
+    auto y = info[2].As<Napi::Number>().Uint32Value();
+    auto z = info[3].As<Napi::Number>().Uint32Value();
+    auto w = info[4].As<Napi::Number>().Uint32Value();
```

Comment:
use `v0`, `v1`, `v2` and `vN` instead of the `x`, `y`, `z`, this method doesn't expect the order: xyzw.